R/s3.put.R

```diff
@@ -27,13 +27,16 @@ s3.put <- function (x, path, name, bucket.location = "US", verbose = FALSE,
       ifelse(verbose, "--verbose --progress", "--no-progress"), ifelse(debug,
           "--debug", ""), '--check-md5')
 
-  run_system_put(path, name, s3.cmd, check_exists, num_retries)
+  ## Ensure backoff vector has correct number of elements and is capped
+  backoff <- backoff[vapply(1:num_retries, function(i) min(i, length(backoff)), integer(1))]
```

Comment:
Just added an if statement because num_retries is an integer.  1:0 doesn't break anything but it's just nonsensical in this context because backoff is never used.